styx-api-service/src/test/java/com/spotify/styx/api/ResourceResourceTest.java

```diff
@@ -0,0 +1,257 @@
+
+    // make sure both are listed
+    Response<ByteString> listResponse =
+        awaitResponse(serviceHelper.request("GET", path("")));
+    assertThat(listResponse, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+    assertJson(listResponse, "resources", hasSize(2));
+    assertJson(listResponse, "resources", containsInAnyOrder(
+        ImmutableMap.of("id", "resource1", "concurrency", 1),
+        ImmutableMap.of("id", "resource2", "concurrency", 2)
+    ));
+
+    // change resource2
+    Response<ByteString> putResponse =
+        awaitResponse(serviceHelper.request("PUT", path("/resource2"),
+            ByteString.encodeUtf8("{\"id\": \"resource2\", \"concurrency\": 3}")));
+    assertThat(putResponse, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+    assertJson(putResponse, "id", equalTo("resource2"));
+    assertJson(putResponse, "concurrency", equalTo(3));
+
+    // make sure resource2 is changed in list and that resource1 is unchanged
+    Response<ByteString> listResponse2 =
+        awaitResponse(serviceHelper.request("GET", path("")));
+    assertThat(listResponse2, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+    assertJson(listResponse2, "resources", hasSize(2));
+    assertJson(listResponse2, "resources", containsInAnyOrder(
+        ImmutableMap.of("id", "resource1", "concurrency", 1),
+        ImmutableMap.of("id", "resource2", "concurrency", 3)
+    ));
+    assertThat(storage.resource(RESOURCE_1.id()).get(), is(RESOURCE_1));
+    assertThat(storage.resource("resource2").get(), is(Resource.create("resource2", 3)));
+
+    // delete resource2
+    Response<ByteString> deleteResponse =
+        awaitResponse(serviceHelper.request("DELETE", path("/resource2")));
+    assertThat(deleteResponse, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+
+    // make sure resource2 is not showing up in the listing
+    Response<ByteString> listResponse3 =
+        awaitResponse(serviceHelper.request("GET", path("")));
+    assertThat(listResponse3, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+    assertJson(listResponse3, "resources", hasSize(1));
+    assertJson(listResponse3, "resources", containsInAnyOrder(
+        ImmutableMap.of("id", "resource1", "concurrency", 1)
+    ));
+    assertThat(storage.resource(RESOURCE_1.id()).get(), is(RESOURCE_1));
+    assertThat(storage.resource("resource2").isPresent(), is(false));
+  }
+
+  private Response<ByteString> awaitResponse(CompletionStage<Response<ByteString>> completionStage)
+      throws InterruptedException, java.util.concurrent.ExecutionException,
+             java.util.concurrent.TimeoutException {
+    return completionStage.toCompletableFuture().get(5, TimeUnit.SECONDS);
+  }
+
+  private String path(String path) {
+    return Api.Version.V1.prefix() + ResourceResource.BASE + path;
+  }
+
+  private Connection setupBigTableMockTable() {
```

Comment:
Do we have to set up mocked behaviour for bigtable for this test? I think it would be enough just to send in a plain mock, since none of the paths will interact with bigtable.